File: collie/src/main/java/com/snail/collie/fps/FpsThread.java
package com.snail.collie.fps;

import java.util.concurrent.LinkedBlockingQueue;

public class FpsThread extends Thread {

    private LinkedBlockingQueue<Runnable> mLinkedBlockingQueue;

    public FpsThread(LinkedBlockingQueue<Runnable> queue) {
        super("fps_track");
        mLinkedBlockingQueue = queue;
    }

    @Override
    public void run() {
        super.run();
        while (true) {
            try {
                Runnable runnable = mLinkedBlockingQueue.take();
                runnable.run();
            } catch (Exception ignored) {
            }
        }

    }
}

File: collie/src/main/java/com/snail/kotlin/fps/FpsThread.kt
package com.snail.kotlin.fps

import java.util.concurrent.LinkedBlockingQueue

/**
 * Author: snail
 * Data: 2021/10/9.
 * Des:
 * version:
 */
class FpsThread(var linkedBlockingQueue: LinkedBlockingQueue<Runnable>?) : Thread() {

    override fun run() {
        super.run()
        while (true) {
            try {
                val runnable = linkedBlockingQueue?.take()
                runnable?.run()
            } catch (ignored: Exception) {
            }
        }
    }
}